stac_api/infrastructure/construct.py

```diff
@@ -71,12 +71,23 @@ def __init__(
         )
 
         lambda_function.add_environment(
-            "VEDA_STAC_PATH_PREFIX", veda_stac_settings.path_prefix
+            "VEDA_STAC_PATH_PREFIX", veda_stac_settings.stac_path_prefix
         )
 
         stac_api_integration = (
             aws_apigatewayv2_integrations_alpha.HttpLambdaIntegration(
-                construct_id, handler=lambda_function
+                construct_id,
+                handler=lambda_function,
+                parameter_mapping=(
+                    aws_apigatewayv2_alpha.ParameterMapping().overwrite_header(
+                        "host",
+                        aws_apigatewayv2_alpha.MappingValue(
+                            veda_stac_settings.domain_hosted_zone_name
+                        )
+                        if veda_stac_settings.domain_hosted_zone_name
+                        else None,
+                    )
+                ),
```

Comment:
@smohiudd I realized we'd want to do this to the raster API too, because otherwise when we use `/mosaic/register`, it gives back the API gateway URL

Reply:
Thanks @slesaad have you tested that on GHGC yet?

Reply:
yes! - https://github.com/US-GHG-Center/ghgc-backend/pull/46/files
deployed to dev.ghg.center/api/raster